src/function/special/riemannZeta.js

```diff
@@ -0,0 +1,87 @@
+import { factory } from '../../utils/factory.js'
+
+const name = 'riemannZeta'
+const dependencies = ['typed', 'config', 'multiply', 'pow', 'divide', 'factorial', 'gamma', 'sin', 'subtract', 'add', 'Complex']
+
+export const createRiemannZeta = /* #__PURE__ */ factory(name, dependencies, ({ typed, config, multiply, pow, divide, factorial, gamma, sin, subtract, add, Complex }) => {
+  /**
+   * Compute the Riemann Zeta function of a value using an infinite series for
+   * all of the complex plane using Riemann's Functional equation.
+   *
+   *
+   * Syntax:
+   *
+   *    math.riemannZeta(n)
+   *
+   * Examples:
+   *  FIX::
+   *    math.riemannZeta(5)       // returns 1.0369277551433699...
+   *    math.riemannZeta(-0.5)    // returns -0.2078862249773...
+   *    math.riemannZeta(math.i)  // returns 0.0033002..., -0.4181554491...i
+   *
+   *
+   * @param {number | Complex} z   A real or complex number
+   * @return {number | Complex}    The zeta of `z`
+   */
+  function zeta (s) {
+    s = Complex(s)
```

Comment:
I'm not a fan of replacing your input arguments. How about moving the `s = new Complex(s)` out, like:

```js
return typed(name, {
  number: z => zeta(new Complex(z)),
  Complex: zeta
})
```